Improve formatting of Player HUD flags documentation

ClobberXD opened the pull request:
See https://github.com/minetest/minetest/pull/7582#issuecomment-407791483 for details.

ClobberXD commented on doc/lua_api.txt:
```diff
@@ -6178,7 +6175,7 @@ Used by `minetest.create_detached_inventory`.
 HUD Definition
 --------------
 
-Used by `minetest.hud_add`. Returned by `minetest.hud_get`.
```
Introduced in ffdb1ba6

@pauloue

paramat commented on doc/lua_api.txt:
```diff
@@ -4938,18 +4938,15 @@ This is basically a reference to a C++ `ServerActiveObject`
-* `hud_get_flags()`: returns a table containing status of hud flags
-    * returns `{hotbar=true, healthbar=true, crosshair=true, wielditem=true,
-      breathbar=true, minimap=true, minimap_radar=true}`
+* `hud_get_flags()`: returns a table containing the player's HUD flags.
```
Needs to specify it is a table of bools for the flags.

paramat commented on doc/lua_api.txt:
```diff
@@ -6201,6 +6198,21 @@ Used by `minetest.hud_add`. Returned by `minetest.hud_get`.
     --  ^ Size of element in pixels
     }
+    
+Player HUD Flags
```
Not sure about putting this in definition tables, as it seems out of place, how about in the earlier HUD section?

paramat replied:
Seems out of place here, it is a table of parameters for an API, not a definition table.

paramat commented on doc/lua_api.txt:
```diff
@@ -4938,18 +4938,15 @@ This is basically a reference to a C++ `ServerActiveObject`
-      unset.
+* `hud_set_flags(flags)`: sets specified HUD flags of player.
+    * A table containing a `true`/`false` value of the HUD flags to be set or
+      unset. See `Player HUD Flags` section for a list of possible flags.
```
I actually think it's best to list the table entries here instead of elsewhere (which is annoying to find).

ClobberXD replied:
Agreed. This table is only relevant to `hud_get_flags` and `hud_set_flags`, which are right next to each other...

paramat commented:
I think this section can be improved, but leaving the list of table entries where it is currently.

pauloue commented on doc/lua_api.txt:
```diff
@@ -4938,18 +4938,15 @@ This is basically a reference to a C++ `ServerActiveObject`
-    * returns `{hotbar=true, healthbar=true, crosshair=true, wielditem=true,
-      breathbar=true, minimap=true, minimap_radar=true}`
+* `hud_get_flags()`: returns a table containing the player's HUD flags.
+	* See `Player HUD Flags` section for a list of possible flags.
```
This should be 4 spaces. no tabs.

ClobberXD replied:
Thanks for pointing that out, fixed

ClobberXD commented:
Addressed all the short-comings. Anything else left to do?

paramat commented:
For consistency with nearby documentation, i think the list of table fields is best in the 'set' section, then the 'get' section refers to the 'set' section.
The 'set' section is unclear whether this is a table of bools only or has fields set to bools, i think it needs to clarify.

So it could read something like:
```
'hud_set_flags(flags)': sets the visibility of HUD elements for the player.
'flags' is a table with the following fields set to true/false. For example:
{hotbar = true, healthbar = true, crosshair = true, wielditem = true,
breathbar = true, minimap = true, minimap_radar = true}
...
...
hud_get_flags(): returns the table specified in 'hud_set_flags', containing
the status of the player's HUD flags.
```
The previous documentation used true/false instead of 'bool'.

Will tag @Ezhh for opinion as she is our docs expert.

ClobberXD commented:
> sets the visibility of HUD elements for the player

It feels incorrect to me. These flags only set the visibility of the **builtin** HUD elements, and custom HUD elements are unaffected. Hmm... would an additional bool field to disable all **non-builtin** HUD elements be useful?

paramat commented:
Sure, specifying these are the standard HUD elements is fine.

SmallJoker commented on doc/lua_api.txt:
```diff
@@ -4938,18 +4938,23 @@ This is basically a reference to a C++ `ServerActiveObject`
-    * returns `{hotbar=true, healthbar=true, crosshair=true, wielditem=true,
-      breathbar=true, minimap=true, minimap_radar=true}`
+* `hud_get_flags()`: returns a table of booleans containing the player's HUD flags.
+        * {
```
This is not displayed correctly in a markdown viewer.

SmallJoker commented on doc/lua_api.txt:
```diff
@@ -4938,18 +4938,23 @@ This is basically a reference to a C++ `ServerActiveObject`
-* `hud_get_flags()`: returns a table containing status of hud flags
-    * returns `{hotbar=true, healthbar=true, crosshair=true, wielditem=true,
-      breathbar=true, minimap=true, minimap_radar=true}`
+* `hud_get_flags()`: returns a table of booleans containing the player's HUD flags.
```
Suggestion: `a table of player flags with boolean as value`

ClobberXD replied:
`player HUD flags`?

paramat replied:
Better it refers to the table in the 'set' section to be specific, otherwise it may be misunderstood as {bool, bool, bool ...}.

SmallJoker commented on doc/lua_api.txt:
```diff
@@ -6201,7 +6206,7 @@ Used by `minetest.hud_add`. Returned by `minetest.hud_get`.
     --  ^ Size of element in pixels
     }
-
+    
```
Superfluous spaces

pauloue commented on doc/lua_api.txt:
````diff
@@ -4938,18 +4938,25 @@ This is basically a reference to a C++ `ServerActiveObject`
+    * ```
+      {
+           hotbar,
+           healthbar,
````
I don't think a code block is necessary here. Just add `Possible flags:` to the prev line and use a list:
```
* hotbar
* healthbar
```

ClobberXD commented:
Converted the HUD flags table into a list and moved it into `hud_set_flags`. Anything else left to do? Is the description of the two methods good enough?

https://github.com/ClobberXD/minetest/blob/c915f3cfb014517617c80147bdaf36eebd0e84ba/doc/lua_api.txt#L4941-L4942

https://github.com/ClobberXD/minetest/blob/c915f3cfb014517617c80147bdaf36eebd0e84ba/doc/lua_api.txt#L4955-L4956

SmallJoker commented on doc/lua_api.txt:
```diff
@@ -4938,18 +4938,22 @@ This is basically a reference to a C++ `ServerActiveObject`
+* `hud_set_flags(flags)`: sets specified HUD flags of player.
+    * `flags`: A table of player HUD flags with boolean values.
+    * List of flags that can be toggled:
+      * hotbar
```
Each indent level should have the same amount of spaces. These flags need two more.

ClobberXD replied:
Thanks for spotting that one; fixed.

paramat commented on doc/lua_api.txt:
```diff
@@ -4938,18 +4938,22 @@ This is basically a reference to a C++ `ServerActiveObject`
-      breathbar=true, minimap=true, minimap_radar=true}`
+* `hud_set_flags(flags)`: sets specified HUD flags of player.
+    * `flags`: A table of player HUD flags with boolean values.
+    * List of flags that can be toggled:
```
This is still no clearer than before. This line and the previous line could be replaced with:
`flags` is a table with the following fields set to boolean values:
It's already known these are HUD flags so no need to mention that again. See nearby docs above, the word 'fields' is used instead which clarifies 'thing = value'.
```

 * `get_player_control()`: returns table with player pressed keys
* The table consists of fields with boolean value representing the pressed
```
```
 * `set_physics_override(override_table)`
* `override_table` is a table with the following fields:
```

paramat replied:
+1 if these 2 lines changed to:
```
`flags` is a table with the following fields set to boolean values:
```

ClobberXD replied:
Isn't adding a `;` after the param name, the de facto standard? Should I squash the two lines into the following instead?
```
`flags`: A table with the following fields set to boolean values
```

paramat replied:
That would be fine +1.

ClobberXD replied:
Done

paramat commented on doc/lua_api.txt:
```diff
@@ -4938,18 +4938,22 @@ This is basically a reference to a C++ `ServerActiveObject`
+    * `minimap`: Modifies the client's permission to view the minimap.
+      The client may locally elect to not view the minimap.
+    * `minimap_radar` is only usable when `minimap` is true
+* `hud_get_flags()`: returns a table of player HUD flags with boolean values.
```
It's clearest to explain the form is identical to that used in 'set flags'. Like:
`'get_physics_override()': returns the table given to 'set_physics_override'`

ClobberXD replied:
`get_physics_override` returns exactly the same table that was previously passed to `set_physics_override`. In the case of `hud_get_flags` however, the table returned is _not _ the same as the table passed to it's corresponding setter, but is one with _all_ the HUD flags and their respective boolean values. Also, the table passed to `hud_set_flags` allows fields with `nil` values - i.e. omitted fields.

paramat replied:
Good point.

paramat replied:
Ok this line is ok for me.

ClobberXD replied:
Done
